dashboard/test/factories/pd_factories.rb

```diff
@@ -452,11 +452,20 @@
           create :pd_attendance, session: session, teacher: teacher
         end
       end
-      if evaluator.cdo_scholarship_recipient
+
+      scholarship_params = {
+        user: teacher,
+        course: Pd::Workshop::COURSE_KEY_MAP[evaluator.workshop.course],
+        application_year: evaluator.workshop.school_year
+      }
+
+      # We have an after_create hook on the enrollment model (see :set_default_scholarship_info)
+      # that creates a ScholarshipInfo entry in certain cases (namely, if the enrollment is in a CSF workshop).
+      # Skip creating a new ScholarshipInfo entry if one has already been created
+      # when the enrollment is created above.
+      if evaluator.cdo_scholarship_recipient && Pd::ScholarshipInfo.find_by(scholarship_params).nil?
```

Comment:
nit: would it be shorter to use https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by?

Reply:
Would I be able to use the `pd_scholarship_info` factory that way?

Reply:
I guess in this case I'm not making much use of the factory anyway.